pxf/pxf-jdbc/src/main/java/org/apache/hawq/pxf/plugins/jdbc/JdbcPartitionFragmenter.java

```diff
@@ -0,0 +1,284 @@
+ * </pre>
+ * The <code>PARTITION_BY</code> parameter can be split by colon(':'),the <code>column_type</code> current supported : <code>date,int,enum</code> .
+ * The Date format is 'yyyy-MM-dd'. <p>
+ * The <code>RANGE</code> parameter can be split by colon(':') ,used to identify the starting range of each fragment.
+ * The range is left-closed, ie:<code> '>= start_value AND < end_value' </code>.If the <code>column_type</code> is <code>int</code>,
+ * the <code>end_value</code> can be empty. If the <code>column_type</code>is <code>enum</code>,the parameter <code>RANGE</code> can be empty. <p>
+ * The <code>INTERVAL</code> parameter can be split by colon(':'), indicate the interval value of one fragment.
+ * When <code>column_type</code> is <code>date</code>,this parameter must be split by colon, and <code>interval_unit</code> can be <code>year,month,day</code>.
+ * When <code>column_type</code> is <code>int</code>, the <code>interval_unit</code> can be empty.
+ * When <code>column_type</code> is <code>enum</code>,the <code>INTERVAL</code> parameter can be empty.
+ * </p>
+ * <p>
+ * The syntax examples is :<p>
+ * <code>PARTITION_BY=createdate:date&RANGE=2008-01-01:2010-01-01&INTERVAL=1:month'</code> <p>
+ * <code>PARTITION_BY=year:int&RANGE=2008:2010&INTERVAL=1</code> <p>
+ * <code>PARTITION_BY=grade:enum&RANGE=excellent:good:general:bad</code>
+ * </p>
+ *
+ */
+public class JdbcPartitionFragmenter extends Fragmenter {
+    String[] partition_by = null;
+    String[] range = null;
+    String[] interval = null;
+    PartitionType partitionType = null;
+    String partitionColumn = null;
+    IntervalType intervalType = null;
+    int intervalNum = 1;
+
+    enum PartitionType {
+        DATE,
+        INT,
+        ENUM;
+
+        public static PartitionType getType(String str) {
+            return valueOf(str.toUpperCase());
+        }
+    }
+
+    enum IntervalType {
+        DAY,
+        MONTH,
+        YEAR;
+
+        public static IntervalType type(String str) {
+            return valueOf(str.toUpperCase());
+        }
+    }
+
+    //The unit interval, in milliseconds, that is used to estimate the number of slices for the date partition type
+    static Map<IntervalType, Long> intervals = new HashMap<IntervalType, Long>();
+
+    static {
+        intervals.put(IntervalType.DAY, (long) 24 * 60 * 60 * 1000);
+        intervals.put(IntervalType.MONTH, (long) 30 * 24 * 60 * 60 * 1000);//30 day
+        intervals.put(IntervalType.YEAR, (long) 365 * 30 * 24 * 60 * 60 * 1000);//365 day
+    }
+
+    /**
+     * Constructor for JdbcPartitionFragmenter.
+     *
+     * @param inConf input data such as which Jdbc table to scan
+     * @throws JdbcFragmentException
+     */
+    public JdbcPartitionFragmenter(InputData inConf) throws JdbcFragmentException {
+        super(inConf);
+        if(inConf.getUserProperty("PARTITION_BY") == null )
+            return;
+        partition_by = inConf.getUserProperty("PARTITION_BY").split(":");

```

Comment:
since all of these properties are user provided, maybe check that they match the expectations as specified in the documentation above, and if not provide a user friendly error message. `UserDataException` might be appropriate.